main.tf

```diff
@@ -130,6 +130,13 @@ locals {
     byte_limit    = var.byte_limit
     record_limit  = var.record_limit
     time_limit_ms = var.time_limit_ms
+
+    disable           = !tobool(var.telemetry_enabled)
+    telemetry_url     = module.telemetry.0.collector_uri
```

Comment:
This will fail if telemetry is disabled - you need something like this:

```
join("", module.telemetry.*.collector_uri)
```

This uses the splat operator and a join to allow you to circumvent the optionality of this output

Reply:
Done the same to `auto_generated_id = module.telemetry.0.auto_generated_id` too